compiler-rt/lib/hwasan/hwasan_report.cpp

```diff
@@ -585,6 +585,13 @@ void BaseReport::PrintAddressDescription() const {
         untagged_addr - heap.begin, d.Default());
   }
 
+  auto announce_by_id = [](u32 thread_id) {
```

Comment:
I am not a fan, I don't think this indirection adds anything.

Reply:
we used it twice
so either repeat or indirection